examples/arm/executor_runner/arm_executor_runner.cpp

```diff
@@ -135,6 +135,44 @@ void et_pal_emit_log_message(
 
 namespace {
 
+// Setup our own allocator that can show some extra stuff like used and free
+// memory info
+class ArmMemoryAllocator : public executorch::runtime::MemoryAllocator {
+ public:
+  ArmMemoryAllocator(uint32_t size, uint8_t* base_address)
+      : MemoryAllocator(size, base_address), used_(0) {}
+
+  void* allocate(size_t size, size_t alignment = kDefaultAlignment) override {
+    void* ret = executorch::runtime::MemoryAllocator::allocate(size, alignment);
+    if (ret != nullptr) {
+      // Align with the same code as in MemoryAllocator::allocate() to keep
+      // used_ "in sync" As alignment is expected to be power of 2 (checked by
+      // MemoryAllocator::allocate()) we can check it the lower bits
+      // (same as alignment - 1) is zero or not.
+      if ((size & (alignment - 1)) == 0) {
+        // Already aligned.
+        used_ += size;
+      } else {
+        used_ = (used_ | (alignment - 1)) + 1 + size;
+      }
+    }
+    return ret;
+  }
+
+  // Returns the used size of the allocator's memory buffer.
+  size_t used_size() const {
+    return used_;
+  }
+
+  // Returns the free size of the allocator's memory buffer.
+  size_t free_size() const {
+    return executorch::runtime::MemoryAllocator::size() - used_;
+  }
```

Comment:
can you add tests for these?

Reply:
As this is in our user example code I tried to keep it clean but it would be really nice to have some test on this code in some way Ill have it the back of my mind for maturing a bit.